pyked/converters.py

```diff
@@ -0,0 +1,522 @@
+        raise MissingElementError('bibliographyLink')
+
+    # Try to get reference info via DOI, fall back on preferredKey if necessary.
+    ref_doi = elem.get('doi', None)
+    ref_key = elem.get('preferredKey', None)
+
+    if ref_doi is not None:
+        try:
+            ref = habanero.Crossref().works(ids=ref_doi)['message']
+        except (HTTPError, habanero.RequestError, ConnectionError, UnboundLocalError):
+            if ref_key is None:
+                raise KeywordError('DOI not found and preferredKey attribute not set')
+            else:
+                warn('Missing doi attribute in bibliographyLink or lookup failed. '
+                     'Setting "detail" key as a fallback; please update to the appropriate fields.'
+                     )
+                reference['detail'] = ref_key
+                if reference['detail'][-1] != '.':
+                    reference['detail'] += '.'
+        else:
+            if ref_key is not None:
+                warn('Using DOI to obtain reference information, rather than preferredKey.')
+            reference['doi'] = elem.attrib['doi']
+            # Now get elements of the reference data
+            # Assume that the reference returned by the DOI lookup always has a container-title
+            reference['journal'] = ref.get('container-title')[0]
+            ref_year = ref.get('published-print') or ref.get('published-online')
+            reference['year'] = int(ref_year['date-parts'][0][0])
+            reference['volume'] = int(ref.get('volume'))
+            reference['pages'] = ref.get('page')
+            reference['authors'] = []
+            for author in ref['author']:
+                auth = {}
+                auth['name'] = ' '.join([author['given'], author['family']])
+                # Add ORCID if available
+                orcid = author.get('ORCID')
+                if orcid:
+                    auth['ORCID'] = orcid.lstrip('http://orcid.org/')
+                reference['authors'].append(auth)
+
+    elif ref_key is not None:
+        warn('Missing doi attribute in bibliographyLink. '
+             'Setting "detail" key as a fallback; please update to the appropriate fields.'
+             )
+        reference['detail'] = ref_key
+        if reference['detail'][-1] != '.':
+            reference['detail'] += '.'
+    else:
+        # Need one of DOI or preferredKey
+        raise MissingAttributeError('preferredKey', 'bibliographyLink')
+
+    return reference
+
+
+def get_experiment_kind(root):
+    """Read common properties from root of ReSpecTh XML file.
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with experiment type and apparatus information.
+    """
+    properties = {}
+    if root.find('experimentType').text == 'Ignition delay measurement':
+        properties['experiment-type'] = 'ignition delay'
+    else:
+        #TODO: support additional experimentTypes
```

Comment:
I suppose that adding additional experiments would require a major overhaul of this module, so keeping track of this one place isn't that important...